client/network/src/service.rs

```diff
@@ -675,9 +675,8 @@ where
 		self.service.remove_reserved_peer(peer);
 	}
 
-	/// Adds a `PeerId` and its address as reserved. The string should encode the address
-	/// and peer ID of the remote node.
-	pub fn add_reserved_peer(&self, peer: String) -> Result<(), String> {
+	/// Adds a `PeerId` and its address as reserved. 
+	pub fn add_reserved_peer(&self, peer: Multiaddr) -> Result<(), String> {
```

Comment:
I through the main reason of this PR was to accept MultiaddrWithPeerId.

Reply:
Initially, but since the issue it aims to resolve gives the option to accept either Multiaddr or MultiaddrWithPeerId, I thought it may be cleaner to opt for Multiaddr instead.